File: fixtures/scale/java/large/src/cryptobench/ciphers/xtea/XteaSuite.java
package cryptobench.ciphers.xtea;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class XteaSuite implements CipherSuite {
    @Override
    public String name() {
        return "xtea";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new XteaCipher(XteaKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : XteaVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/xtea/XteaSuite.kt
package cryptobench.ciphers.xtea

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class XteaSuite : CipherSuite {
    override fun name(): String = "xtea"

    override fun run(): SuiteResult {
        val cipher: Cipher = XteaCipher(XteaKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in XteaVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
